File: server/models/subtransaction.rb
DB.create_table? :subtransactions do
  uuid :uuid, primary_key: true

  String :description
  # Amount in cents
  Integer :amount, null: false, default: 0
  foreign_key :category_uuid, :categories, type: 'uuid'
  foreign_key :transaction_uuid, :transactions, on_delete: :cascade, type: 'uuid'
  DateTime :created_at, null: false
  DateTime :updated_at, null: false
end

class Subtransaction < Sequel::Model
  many_to_one :category, key: :category_uuid
  many_to_one :transaction, key: :transaction_uuid
end

File: packages/server/src/database/entities/Subtransaction.js
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm'
import {Category} from './Category'
import {Transaction} from './Transaction'

@Entity()
export class Subtransaction {
  @PrimaryGeneratedColumn('uuid')
  uuid = undefined

  @Column('varchar', {nullable: true})
  description = undefined

  @Column('integer')
  amount = undefined

  @Column('uuid')
  category_uuid = undefined
  @ManyToOne(() => Category)
  @JoinColumn({name: 'category_uuid'})
  category = undefined

  @ManyToOne(() => Transaction, transaction => transaction.subtransactions, {
    nullable: false,
  })
  transaction = undefined

  @CreateDateColumn()
  created_at = undefined

  @UpdateDateColumn()
  updated_at = undefined
}
